File: android/app/src/main/java/com/lilru/liftr/ui/pets/PetMarketPurchaseFeedback.kt
package com.lilru.liftr.ui.pets

import com.lilru.liftr.ui.AppSnackbar
import kotlinx.coroutines.flow.MutableStateFlow
import kotlinx.coroutines.flow.StateFlow
import kotlinx.coroutines.flow.asStateFlow
import kotlinx.coroutines.flow.update

object PetMarketPurchaseFeedback {
    private val _unseenMyItemsCount = MutableStateFlow(0)
    val unseenMyItemsCount: StateFlow<Int> = _unseenMyItemsCount.asStateFlow()

    fun recordPurchase(itemType: String) {
        if (itemType != "pet_egg" && itemType != "incubator") return
        _unseenMyItemsCount.update { it + 1 }
        val message = when (itemType) {
            "pet_egg" -> "Mysterious Egg added to My Items"
            else -> "Egg Incubator added to My Items"
        }
        AppSnackbar.showSuccess(message)
    }

    fun clearBadge() {
        _unseenMyItemsCount.value = 0
    }
}

File: Liftr/Pet/PetMarketPurchaseFeedback.swift
import Foundation
import Combine

@MainActor
final class PetMarketPurchaseFeedback: ObservableObject {
    static let shared = PetMarketPurchaseFeedback()

    @Published private(set) var unseenMyItemsCount = 0
    @Published var toastMessage: String?

    private var toastDismissTask: Task<Void, Never>?

    private init() {}

    func recordPurchase(itemType: String) {
        guard itemType == "pet_egg" || itemType == "incubator" else { return }
        unseenMyItemsCount += 1
        toastMessage = itemType == "pet_egg"
            ? "Mysterious Egg added to My Items"
            : "Egg Incubator added to My Items"
        toastDismissTask?.cancel()
        toastDismissTask = Task {
            try? await Task.sleep(nanoseconds: 3_000_000_000)
            toastMessage = nil
        }
    }

    func clearBadge() {
        unseenMyItemsCount = 0
    }
}
